sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/FeatureManager.java

```diff
@@ -44,83 +73,352 @@ public class FeatureManager {
+        return checkFeature(feature, featureContext).block().getVariant();
+    }
+
+    /**
+     * Returns the variant assigned to the current context.
+     * 
+     * @param feature Feature being checked.
+     * @return Assigned Variant
+     */
+    public Mono<Variant> getVariantAsync(String feature) {
+        return checkFeature(feature, null).map(event -> event.getVariant());
+    }
+
+    /**
+     * Returns the variant assigned to the current context.
+     * 
+     * @param feature Feature being checked.
+     * @param featureContext Local context
+     * @return Assigned Variant
+     */
+    public Mono<Variant> getVariantAsync(String feature, Object featureContext) {
+        return checkFeature(feature, featureContext).map(event -> event.getVariant());
+    }
+
+    private Mono<EvaluationEvent> checkFeature(String featureName, Object featureContext)
+        throws FilterNotFoundException {
+        Feature featureFlag = featureManagementConfigurations.getFeatureFlags().stream()
+            .filter(feature -> feature.getId().equals(featureName)).findAny().orElse(null);
+
+        EvaluationEvent event = new EvaluationEvent(featureFlag);
+
+        if (featureFlag == null) {
+            LOGGER.warn("Feature flag %s not found", featureName);
+            return Mono.just(event);
         }
 
-        Boolean boolFeature = featureManagementConfigurations.getOnOff().get(feature);
+        if (!featureFlag.isEnabled()) {
+            this.assignDefaultDisabledVariant(event);
+            if (featureFlag.getAllocation() != null) {
+                String variantName = featureFlag.getAllocation().getDefaultWhenDisabled();
+                event.setVariant(this.variantNameToVariant(featureFlag, variantName));
+            }
 
-        if (boolFeature != null) {
-            return boolFeature;
+            // If a feature flag is disabled and override can't enable it
+            return Mono.just(event.setEnabled(false));
         }
 
-        Feature featureItem = featureManagementConfigurations.getFeatureManagement().get(feature);
+        Mono<EvaluationEvent> result = this.checkFeatureFilters(event, featureContext);
 
-        if (featureItem == null || !featureItem.getEvaluate()) {
-            return false;
+        result = assignAllocation(result, context);
+        return result;
+    }
+
+    private Mono<EvaluationEvent> assignAllocation(Mono<EvaluationEvent> monoEvent, Object featureContext) {
+
+        return monoEvent.map(event -> {
+
+            Feature featureFlag = event.getFeature();
+
+            if (featureFlag.getVariants() == null || featureFlag.getAllocation() == null) {
+                return event;
+            }
+
+            if (!event.isEnabled()) {
+                this.assignDefaultDisabledVariant(event);
+                event.setVariant(
+                    this.variantNameToVariant(featureFlag, featureFlag.getAllocation().getDefaultWhenDisabled()));
+                return event;
```

Comment:
To avoid status override when the event is disabled, should we explicitly have `event.setEnabled(false)` here?